infrastructure/src/main/java/org/corfudb/infrastructure/ServerRequestHandler.java

```diff
@@ -0,0 +1,162 @@
+
+    }
+
+    @Override
+    protected void handleCompactLog(Request request, ChannelHandlerContext ctx) {
+
+    }
+
+    @Override
+    protected void handleExecuteWorkFlow(Request request, ChannelHandlerContext ctx) {
+
+    }
+
+    @Override
+    protected void handleFlash(Request request, ChannelHandlerContext ctx) {
+
+    }
+
+    @Override
+    protected void handleGetLayout(Request request, ChannelHandlerContext ctx) {
+
+    }
+
+    @Override
+    protected void handleGetToken(Request request, ChannelHandlerContext ctx) {
+
+    }
+
+    @Override
+    protected void handleHealFailure(Request request, ChannelHandlerContext ctx) {
+
+    }
+
+    @Override
+    protected void handlePrepareLayout(Request request, ChannelHandlerContext ctx) {
+
+    }
+
+    @Override
+    protected void handleProposeLayout(Request request, ChannelHandlerContext ctx) {
+
+    }
+
+    @Override
+    protected void handleQueryLogMetadata(Request request, ChannelHandlerContext ctx) {
+
+    }
+
+    @Override
+    protected void handleQueryNode(Request request, ChannelHandlerContext ctx) {
+
+    }
+
+    @Override
+    protected void handleQueryStream(Request request, ChannelHandlerContext ctx) {
+
+    }
+
+    @Override
+    protected void handleReadLog(Request request, ChannelHandlerContext ctx) {
+
+    }
+
+    @Override
+    protected void handleReportFailure(Request request, ChannelHandlerContext ctx) {
+
+    }
+
+    @Override
+    protected void handleReset(Request request, ChannelHandlerContext ctx) {
+        log.warn("handleReset: Remote reset requested from client with " +
+                "LSB: {} MSB:{}", request.getHeader().getClientId().getLsb(),
+                request.getHeader().getClientId().getMsb());
+
+        // send ResetResponse message back to the client
+        ByteBuf byteBuf = PooledByteBufAllocator.DEFAULT.buffer();
+        // Temporary marker to indicate new Protobuf Message
+        byteBuf.writeByte(0x2);
+        byteBuf.writeBytes(API.newResetResponse(request.getHeader()).toByteArray());
+        ctx.writeAndFlush(byteBuf);
+
+        CorfuServer.restartServer(true);
+    }
+
+    @Override
+    protected void handleRestart(Request request, ChannelHandlerContext ctx) {
+        log.warn("handleRestart: Remote restart requested from client with " +
+                "LSB: {} MSB:{}", request.getHeader().getClientId().getLsb(),
+                request.getHeader().getClientId().getMsb());
+
+        // send RestartResponse message back to the client
+        ByteBuf byteBuf = PooledByteBufAllocator.DEFAULT.buffer();
+        // Temporary marker to indicate new Protobuf Message
+        byteBuf.writeByte(0x2);
+        byteBuf.writeBytes(API.newRestartResponse(request.getHeader()).toByteArray());
+        ctx.writeAndFlush(byteBuf);
+
+        CorfuServer.restartServer(false);
+    }
+
+    @Override
+    protected void handleSeal(Request request, ChannelHandlerContext ctx) {
```

Comment:
Should this method be synchronized?

Reply:
Yeah, It should. I used this class as a stub for the client-side of changes. @zfrenette Could you please make a note of it? I will change in this branch too.